kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/NetworkServiceImpl.java

```diff
@@ -383,43 +376,40 @@ public NetInterfaceState getState(String interfaceName) throws KuraException {
                     netInterfaces.add(netInterface);
                 }
             } catch (KuraException e) {
-                s_logger.error("Can't get network interface info for {} :: exception - {}", interfaceName,
-                        e.toString());
+                logger.error("Can't get network interface info for {} ", interfaceName, e);
             }
         }
 
         // Return an entry for non-connected modems (those w/o a ppp interface)
-        Iterator<String> it = this.m_addedModems.iterator();
+        Iterator<String> it = this.addedModems.iterator();
         while (it.hasNext()) {
             String modemId = it.next();
-            UsbModemDevice usbModem = this.m_usbModems.get(modemId);
+            UsbModemDevice usbModem = this.usbModems.get(modemId);
             if (usbModem != null) {
                 // only add if there is not already a ppp interface for this modem
                 boolean addModem = true;
                 for (NetInterface<?> netInterface : netInterfaces) {
                     UsbDevice usbDevice = netInterface.getUsbDevice();
-                    if (usbDevice != null) {
-                        if (usbDevice.getUsbPort().equals(usbModem.getUsbPort())) {
-                            addModem = false;
-                            break;
-                        }
+                    if ((usbDevice != null) && usbDevice.getUsbPort().equals(usbModem.getUsbPort())) {
```

Comment:
I think that usbDevice returned by the netInterface.getUsbDevice() can be null. I would keep this for now.